### DISCUSSION TITLE
How to deploy with docker

### DISCUSSION THREAD

**Discussion by HerbertHe** [HerbertHe/iptv-sources#36] [Q&A] [answered]

**Comment by HerbertHe** [comment, marked as answer]
## Deploy with docker

```bash
docker run --name iptv-sources -p 3000:8080 -d herberthe0229/iptv-sources:latest
```

- Run `docker ps` to get container status.

Wait a minute, visit <http://localhost:3000>.

Then, you can use `http://localhost:3000` instead of `https://m3u.ibert.me`.

For example: `https://m3u.ibert.me/cn.m3u` -> `http://localhost:3000/cn.m3u`

Or, you can also deploy with your own server & domain.

## Crontab

Maybe you want to set schedule for auto-updating per 2 hours.

- Download `iptv-update.sh` <https://github.com/HerbertHe/iptv-sources/blob/main/iptv-update.sh> to your homedir.

> If you have used `Environment Variables`, maybe you need to modify the script file.

- Edit you crontab:

```bash
crontab -e
```

- Press keyboard `i` for adding schedule.

- Add:

```cron
0 */2 * * * /bin/sh ~/iptv-update.sh
```

- Press keyboard `ESC` to exit edit mode
- Type `:wq` to save
- Restart crontab service

```bash
service crond restart
```

## Update docker image

### Manually update

- Download bash script file `update-image.sh` <https://github.com/HerbertHe/iptv-sources/blob/main/update-image.sh> to your homedir.

> If you have used `Environment Variables`, maybe you need to modify the script file.

- run

```bash
/bin/sh ~/update-image.sh
```

### Auto update

See <https://github.com/containrrr/watchtower>

**Comment by liyiranlab** [comment]
OMV 计划任务里面添加 /bin/sh ~/iptv-update.sh
无法执行提示如下：
500 - Internal Server Error
Failed to execute command 'export PATH=/bin:/sbin:/usr/bin:/usr/sbin:/usr/local/bin:/usr/local/sbin; export LANG=C.UTF-8; export LANGUAGE=; export SHELL=/bin/sh; sudo --shell --non-interactive --user='root' -- /var/lib/openmediavault/cron.d/userdefined-8ec20adc-d6b8-44da-94a2-ba192ec9b706 2>&1' with exit code '1': /var/lib/openmediavault/cron.d/userdefined-8ec20adc-d6b8-44da-94a2-ba192ec9b706: 35: [[: not found the input device is not a TTY

**Reply by HerbertHe** [reply]
文档更新了，重写部署计划任务

**Comment by keyuanpc** [comment]
(node:8) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
error An unexpected error occurred: "https://registry.npmmirror.com/@rollup/wasm-node/-/wasm-node-4.9.1.tgz: getaddrinfo EAI_AGAIN registry.npmmirror.com".
error https://registry.npmmirror.com/fsevents/-/fsevents-2.3.3.tgz: getaddrinfo EAI_AGAIN registry.npmmirror.com
yarn install v1.22.19
[1/5] Resolving packages...
[2/5] Fetching packages...
info If you think this is a bug, please open a bug report with the information provided in "/app/yarn-error.log".
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.

DOCKER安装完后的日志，其网页打不开。

**Reply by HerbertHe** [reply]
重新跑一次就行了

**Reply by yanzhang4013** [reply]
docker顺利安装， 重启仍然不行

**Comment by leiyou-li** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
你好，我最近正在休假中，无法亲自回复你的邮件。我将在假期结束后，尽快给你回复。

**Comment by airenkuang** [comment]
请问homedir是哪个目录啊，我用的绿联的NAS，finalShell进入后不知道哪个是homedir

**Comment by isxzg** [comment]
换了2个服务器部署完都是用的过程镜像自动停止了，不知道是哪里问题

**Comment by Java2Boy** [comment]
docker 运行出错，logs如下
https://pastebin.com/9D4qBsCg

**Reply by HerbertHe** [reply]
@Java2Boy 不影响，后面移除掉就行了

**Reply by Java2Boy** [reply]
> docker 运行出错，logs如下 https://pastebin.com/9D4qBsCg

文档里没标明hosts模式运行docker ,手动修正后正常运行

**Comment by aniel300** [comment]
is there a docker compose?

**Reply by nbqcw123** [reply]
version: '3.8'
services:
  iptv-sources:
    image: herberthe0229/iptv-sources:latest
    container_name: iptv-sources
    ports:
      - "8016:8080"
    restart: unless-stopped
[Uploading compose.yaml…]()

**Comment by gavinqd** [comment]
Download iptv-update.sh https://github.com/HerbertHe/iptv-sources/blob/main/iptv-update.sh to your homedir.
就是这里搞不明白，下载iptv-update.sh到homedir，具体是哪个位置呀

**Reply by HerbertHe** [reply]
cd ~

**Reply by gavinqd** [reply]
> cd ~

-ash: cd: can't cd to homedir: No such file or directory
是需要新建一个文件夹？？名字叫：homedir

**Reply by HerbertHe** [reply]
home 目录就是 ~ 啊，`cd ~` 就是打开 home

**Reply by gavinqd** [reply]
哈哈哈，好吧，成功了，我放到root文件夹下成功了，谢谢

**Comment by leiyou-li** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
你好，我最近正在休假中，无法亲自回复你的邮件。我将在假期结束后，尽快给你回复。
